### PR TITLE
Bug 2027613: use the correct Alertmanager tenancy proxy

### PR DESCRIPTION
**Pull request by vikram-raj**
**fixes:** https://issues.redhat.com/browse/OCPBUGSM-37644

**Descriptions:**
- fix to use the correct Alertmanager tenancy proxy.
- use the alert manager tenancy endpoint in the dev console for silence alerts.
- poll the silences to silence the alert.

Kapture 2022-04-19 at 00 11 03

### CODE REVIEW

**Comment by jerolimov** [comment]
/uncc
Please assign me again if a review is needed. :)

**Comment by vikram-raj** [comment]
This PR includes changes in the backend package. So, need your review here.

/assign @florkbr

**Comment by invincibleJai** [comment]
/lgtm

Verified the change with the bot and as self provisioner user and I could silence the alert 

alertSilence2

@spadgett / @jhadvig Needed your help with approval, PTAL when you get time. Thanks!!

File: frontend/packages/dev-console/src/components/monitoring/alerts/SilenceAlert.tsx [hunk 1]
```diff
@@ -52,9 +51,10 @@ const SilenceAlert: React.FC<SilenceAlertProps> = ({ rule, namespace }) => {
     if (checked) {
       _.each(rule.silencedBy, (silence) => {
         coFetchJSON
-          .delete(`${ALERTMANAGER_TENANCY_BASE_PATH}/api/v2/silence/${silence.id}`)
+          .delete(
+            `${ALERTMANAGER_TENANCY_BASE_PATH}/api/v2/silence/${silence.id}?namespace=${namespace}`,
```
[the remark this answers is not in the document]
Comment (jhadvig) [reply frontend/packages/dev-console/src/components/monitoring/alerts/SilenceAlert.tsx:55]: also in https://github.com/openshift/console/blob/master/frontend/public/components/notification-drawer.tsx#L250 ?

**Review by jhadvig — COMMENTED** [review]
@vikram-raj thanks for the fix! Looking good. Added two comments. PTAL

File: frontend/packages/dev-console/src/components/monitoring/alerts/SilenceAlert.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (vikram-raj) [reply frontend/packages/dev-console/src/components/monitoring/alerts/SilenceAlert.tsx:55]: we updated the AlertManager URL to use `ALERTMANAGER_TENANCY_BASE_PATH` for only the dev console pages where users have limited access.

**Review by jhadvig — APPROVED** [review]
/approve

**Pull request merged**